content/en/blog/_posts/2022-12-6-provision-volumes-from-cross-namespace-snapshots-alpha.md

```diff
@@ -0,0 +1,133 @@
+* Install the `External Provisioner controller` itself
+* Install the `Container Stroge Interface Driver` itself
+
+## Putting it all together
+
+To see how this works, you can install the sample and try it out.
+This sample do to create PVC in ns1 namespace from VolumeSnapshot in default namespace by using csi-hostpath driver.
+
+1. Deploy VolumeSnapshot controller.
+
+   ```terminal
+   kubectl apply -f https://raw.githubusercontent.com/kubernetes-csi/external-snapshotter/master/client/config/crd/snapshot.storage.k8s.io_volumesnapshotclasses.yaml
+   kubectl apply -f https://raw.githubusercontent.com/kubernetes-csi/external-snapshotter/master/client/config/crd/snapshot.storage.k8s.io_volumesnapshotcontents.yaml
+   kubectl apply -f https://raw.githubusercontent.com/kubernetes-csi/external-snapshotter/master/client/config/crd/snapshot.storage.k8s.io_volumesnapshots.yaml
+
+
+   kubectl apply -f https://raw.githubusercontent.com/kubernetes-csi/external-snapshotter/master/deploy/kubernetes/snapshot-controller/rbac-snapshot-controller.yaml
+   kubectl apply -f https://raw.githubusercontent.com/kubernetes-csi/external-snapshotter/master/deploy/kubernetes/snapshot-controller/setup-snapshot-controller.yaml
+   ```
+
+2. Deploy External Provisioner controller.
+
+   ```terminal
+   kubectl apply -f https://raw.githubusercontent.com/kubernetes-csi/external-provisioner/master/deploy/kubernetes/deployment.yaml 
+   kubectl apply -f https://raw.githubusercontent.com/kubernetes-csi/external-provisioner/master/deploy/kubernetes/rbac.yaml
+   ```
+
+3. Deploy csi-hostpath driver.
+
+   ```terminal
+   cd /tmp
+   git clone --depth=1 https://github.com/kubernetes-csi/csi-driver-host-path.git
+   cd csi-driver-host-path
+   ./deploy/kubernetes-latest/deploy.sh
+   ```
+
+4. Deploy ReferenceGrants CRD
+  
+   ```terminal
+   kubectl apply -f https://raw.githubusercontent.com/kubernetes-sigs/gateway-api/main/config/ crd/experimental/gateway.networking.k8s.io_referencegrants.yaml
+   ```
+
+5. Create StorageClass, PVC, and VolumeSnapshot by the examples in the csi-hostpath repo
+
+    ```terminal
+    cd /tmp/csi-driver-host-path
+    kubectl apply -f examples/csi-storageclass.yaml
+    kubectl apply -f examples/csi-pvc.yaml
+    kubectl apply -f examples/csi-snapshot-v1beta1.yaml
+    ```
+
+6. Ceate a ns1 namespace
+
+    ```terminal
+    kubectl create ns ns1
+    ```
+
+7. Create a ReferenceGrant
+
+   ```yaml
+   apiVersion: gateway.networking.k8s.io/v1alpha2
+   kind: ReferenceGrant
+   metadata:
+     name: bar
+     namespace: default
+   spec:
+     from:
+     - group: ""
+       kind: PersistentVolumeClaim
+       namespace: ns1
+     to:
+     - group: snapshot.storage.k8s.io
+       kind: VolumeSnapshot
+       name: new-snapshot-demo
+   ```
+
+8. Create a PersistentVolumeClaim
+
+   ```yaml
+   apiVersion: v1
+   kind: PersistentVolumeClaim
+   metadata:
+     name: foo-pvc
+     namespace: ns1
+   spec:
+     storageClassName: csi-hostpath-sc
```

Comment:
@sftim 
I have something I want to consult with you.
https://github.com/kubernetes/website/pull/37628/files#diff-3a022011d14568f9dcaaf14b8dce54ea2d72d3a496988e77ce7bb05df60cf6cdR32
This example assumes csi-hostpath. csi-storageclass.yaml also uses exmaples from csi-hostpath repo.
If you change to example, you will not be able to use the above file, so I think it will be a procedure for the user to create a storage class as a premise. Is this alright with you?

Reply:
If we don't change to a different example, we **must** warn readers about the risks of using `hostPath` in production environments.